extensions/amp-carousel/0.2/responsive-attributes.js

```diff
@@ -89,19 +94,26 @@ export class ResponsiveAttributes {
    * @private
    */
   getMediaQueryListsAndValues_(value) {
-    return value.split(',').map(part => {
-      // Find the value portion by looking at the end.
-      const {index} = /[a-z0-9.]+$/.exec(part);
-      const value = part.slice(index);
-      // The media query is everything before the value.
-      const mediaQuery = part.slice(0, index).trim();
-      const mediaQueryList = window.matchMedia(mediaQuery);
-
-      return {
-        mediaQueryList,
-        value,
-      };
-    });
+    return value.split(',')
+        .map(part => {
+          // Find the value portion by looking at the end.
+          const result = /[a-z0-9.]+$/.exec(part);
+          if (!result) {
+            return;
+          }
+
+          const {index} = result;
+          const value = part.slice(index);
+          // The media query is everything before the value.
+          const mediaQuery = part.slice(0, index).trim();
+          const mediaQueryList = window.matchMedia(mediaQuery);
+
+          return {
+            mediaQueryList,
+            value,
+          };
+        })
+        .filter(item => item);
```

Comment:
Superfluous filter?

Reply:
Added a comment on what this is doing.

Reply:
Added a comment to the code explaining why it is there.